src/features/Search/Main.tsx

```diff
@@ -0,0 +1,451 @@
+                        >
+                          View {viewSaved ? 'All' : 'Saved'}
+                        </Button>
+                      )}
+                      <Button onClick={this.downloadData} variant={ButtonVariant.Secondary} isOutlined={true}>
+                        Export Hackers
+                      </Button>
+                    </Box>
+                  </Flex>
+                </Box>
+                <Box>
+                  {this.state.mode === SearchMode.TABLE ? (
+                    <ResultsTable
+                      results={this.filter()}
+                      loading={loading}
+                      userType={account ? account.accountType : UserType.UNKNOWN}
+                    />
+                  ) : (
+                    <StatsComponent
+                      stats={this.state.statsResults}
+                      loading={this.state.loading}
+                      onFilterChange={this.onFilterChange}
+                      existingFilters={this.state.query}
+                    />
+                  )}
+                </Box>
+              </Flex>
+            </Box>
+          </Flex>
+        </Box>
+      </Flex>
+    );
+  }
+  public async componentDidMount() {
+    const account = (await Account.getSelf()).data.data;
+    this.setState({ account });
+
+    if (isSponsor(account)) {
+      const sponsor = (await Sponsor.getSelf()).data.data;
+      this.setState({ sponsor });
+    }
+    await this.triggerSearch();
+  }
+  private getSearchFromQuery(): ISearchParameter[] {
+    const search = getValueFromQuery('q');
+    if (!search) {
+      return [];
+    }
+    try {
+      const searchParam = JSON.parse(search);
+      if (!Array.isArray(searchParam)) {
+        return [];
+      }
+      const isValidSearch =
+        searchParam
+          .map(
+            (value: any): boolean => {
+              return isValidSearchParameter(value);
+            }
+          )
+          .indexOf(false) === -1;
+      return isValidSearch ? searchParam : [];
+    } catch (e) {
+      return [];
+    }
+  }
+
+  private handleSearchModeChanged({ value }: any) {
+    console.log(value);
+    this.setState({ mode: value }, this.triggerSearch);
+  }
+
+  private async triggerStatsSearch(): Promise<void> {
+    try {
+      this.setState({ loading: true });
+      const statsResponse = await Hacker.getStats(this.state.query);
+      const stats: IStats | null = getNestedAttr(statsResponse, [
+        'data',
+        'data',
+        'stats',
+      ]);
+      this.setState({ statsResults: stats, loading: false });
+    } catch (e) {
+      ValidationErrorGenerator(e.data);
+      this.setState({ loading: false });
+    }
+  }
+
+  private getSearchBarFromQuery(): string {
+    const search = getValueFromQuery('searchBar');
+    return search ? decodeURIComponent(search) : '';
+  }
+
+  private downloadData(): void {
+    const headers = [
```

Comment:
not in the scope of this PR but we should probably find a more concise mapping function